packages/sdk/tasks/deposit-bnb.ts

```diff
@@ -0,0 +1,432 @@
+    const depositBobaTx = await L1StandardBridge.depositERC20(
+      L1BobaToken.address,
+      l2BobaTokenAddress,
+      utils.parseEther('10'),
+      99999,
+      '0x'
+    )
+    await depositBobaTx.wait()
+    console.log(`ERC20 deposited - ${depositBobaTx.hash}`)
+
+    console.log('Checking to make sure deposit was successful')
+    // Deposit might get reorged, wait and also log for reorgs.
+    let prevBlockHash: string = ''
+    for (let i = 0; i < 12; i++) {
+      const messageReceipt = await signer.provider!.getTransactionReceipt(
+        depositBobaTx.hash
+      )
+      if (messageReceipt.status !== 1) {
+        console.log(`Deposit failed, retrying...`)
+      }
+
+      // Wait for stability, we want some amount of time after any reorg
+      if (prevBlockHash !== '' && messageReceipt.blockHash !== prevBlockHash) {
+        console.log(
+          `Block hash changed from ${prevBlockHash} to ${messageReceipt.blockHash}`
+        )
+        i = 0
+      } else if (prevBlockHash !== '') {
+        console.log(`No reorg detected: ${i}`)
+      }
+
+      prevBlockHash = messageReceipt.blockHash
+      await sleep(1000)
+    }
+    console.log(`Deposit BOBA confirmed`)
+
+    const l2BobaBalance = await l2Provider.getBalance(l2Signer.address)
+    if (l2BobaBalance.lt(utils.parseEther('10'))) {
+      throw new Error(
+        `bad deposit. recipient balance on L2: ${utils.formatEther(
+          l2BobaBalance
+        )}`
+      )
+    }
+    console.log(`Deposit BOBA success`)
+
+    const depositBnbTx = await l1Signer.sendTransaction({
+      to: OptimismPortal.address,
+      value: utils.parseEther('1'),
+    })
+    await depositBnbTx.wait()
+    console.log(`BNB deposited - ${depositBnbTx.hash}`)
+
+    console.log('Checking to make sure deposit was successful')
+    // Deposit might get reorged, wait and also log for reorgs.
+    prevBlockHash = ''
+    for (let i = 0; i < 12; i++) {
+      const messageReceipt = await signer.provider!.getTransactionReceipt(
+        depositBnbTx.hash
+      )
+      if (messageReceipt.status !== 1) {
+        console.log(`Deposit failed, retrying...`)
+      }
+
+      // Wait for stability, we want some amount of time after any reorg
+      if (prevBlockHash !== '' && messageReceipt.blockHash !== prevBlockHash) {
+        console.log(
+          `Block hash changed from ${prevBlockHash} to ${messageReceipt.blockHash}`
+        )
+        i = 0
+      } else if (prevBlockHash !== '') {
+        console.log(`No reorg detected: ${i}`)
+      }
+
+      prevBlockHash = messageReceipt.blockHash
+      await sleep(1000)
+    }
+    console.log(`Deposit confirmed`)
+
+    const l2BnbBalance = await L2BnbToken.balanceOf(l2Signer.address)
+    if (l2BnbBalance.lt(utils.parseEther('1'))) {
+      throw new Error(
+        `bad deposit. recipient balance on L2: ${utils.formatEther(
+          l2BnbBalance
+        )}`
+      )
+    }
+    console.log(`Deposit BNB success`)
```

Comment:
Same comments here, we don't seem to catch if the transaction truly fails, and we don't look for appropriate balances.